### PR TITLE
Create auto_cherry_pick.yml

### PR DESCRIPTION
**Pull request by Sophie-Xie**
Auto cherry pick bot:

1.If input the cherry pick label (such as "cherry-pick-v3.1"), the PRs with this label will be cherry-picked to target branch (such as v3.1).
2.auto merge (configure true or false)

### CODE REVIEW

File: .github/workflows/auto_cherry_pick.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      runs-on: [self-hosted, nebula-fast]
+      container:
+        image: reg.vesoft-inc.com/vesoft/nebula-dev:centos7
+      steps:
+        - name: auto cherry pick
+          uses: xigongdaEricyang/cherry-pick-robot@with-python
+          with:
+              repo_token: ${{ secrets.GH_BOT_PAT }}
+              pr_label: ${{ github.event.inputs.prlabel }}
+              pr_number: ${{ github.event.pull_request.number }}
```
Comment (xigongdaEricyang) [inline .github/workflows/auto_cherry_pick.yml:25]: pr_num

File: .github/workflows/auto_cherry_pick.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      if: ${{ startsWith(github.event.pull_request.labels.*.name, 'cherry-pick-') && github.event.pull_request.merged == true }}
+      runs-on: [self-hosted, nebula-fast]
+      container:
+        image: reg.vesoft-inc.com/vesoft/nebula-dev:centos7
+      steps:
+        - name: auto cherry pick
+          uses: xigongdaEricyang/cherry-pick-robot@with-python
+          with:
+              repo_token: ${{ secrets.GH_BOT_PAT }}
+              pr_label: ${{ github.event.inputs.prlabel }}
```
Comment (xigongdaEricyang) [inline .github/workflows/auto_cherry_pick.yml:24]: here is regex, not sure if its ok

File: .github/workflows/auto_cherry_pick.yml (hunk 1 above)
Comment (Sophie-Xie) [reply .github/workflows/auto_cherry_pick.yml]: Thanks. I created it last Saturday, I fix it now.

**Review by yixinglu — APPROVED** [review]

File: .github/workflows/auto_cherry_pick.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  auto-cherry-pick:
+    if_merged:
+      if: ${{ startsWith(github.event.pull_request.labels.*.name, 'cherry-pick-') && github.event.pull_request.merged == true }}
+      runs-on: [self-hosted, nebula-fast]
```
Comment (Shylock-Hg) [inline .github/workflows/auto_cherry_pick.yml:16]: I think you could run it on GITHUB hosted runners directly.

File: .github/workflows/auto_cherry_pick.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      container:
+        image: reg.vesoft-inc.com/vesoft/nebula-dev:centos7
```
Comment (Shylock-Hg) [inline .github/workflows/auto_cherry_pick.yml:18]: This docker image is used to build nebula source, I think you could try other thin image like `ubuntu` directly.

File: .github/workflows/auto_cherry_pick.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  auto-cherry-pick:
+      #if: ${{ startsWith(github.event.pull_request.labels.*.name, 'cherry-pick-') && github.event.pull_request.merged == true }}
+      runs-on: ubuntu-latest
+      container:
+        image: reg.vesoft-inc.com/dashboard/dashboard-dev:centos7
```
Comment (Shylock-Hg) [inline .github/workflows/auto_cherry_pick.yml:18]: Why don't you run in `ubuntu-latest` directly?

Comment (xigongdaEricyang) [reply .github/workflows/auto_cherry_pick.yml:18]: nened git version and node version

**Review by xigongdaEricyang — APPROVED** [review]
LGTM

**Pull request merged**